### ISSUE TITLE
Rated games not announced to screen readers - Jam Submission Page

### ISSUE THREAD

**Issue by uxaccess** [itchio/itch.io#1883]
**Describe the bug**
Games aren't announced as rated or unrated to screen readers in the jams submissions' pages.

**To Reproduce**
Pre-conditions: 
- Turn on NVDA (screen reader).
- Have at least one rated game in a jam submissions page.

1. Go to 'https://itch.io/jam/games-for-blind-gamers-5/entries' [Jam Submissions Page with at least one rated game]
2. Press '2' to jump to the h2 heading "Filter Submissions"
3. Tab to the rated game (at least 23 times from "Filter Submissions", if it's the first game on the list)
4. See error - The rated game shows up visually as rated (with a checkmark and the word 'rated'), but isn't announced to the screen reader

**Expected behavior**
The screen reader announces the game as rated.

**Screenshots**
n/a

**Desktop (please complete the following information):**
- OS: Windows 11
- Browser: Chrome
- Version: 145.0.7632.162

**Smartphone (please complete the following information):**
n/a

**Additional context**
We run a Games for Blind Gamers jam, made for completely blind developers and gamers, and have just realized our blind community members don't have an equal experience on this page as our sighted community members. We would like to see this resolved so we can all have a pleasant experience playing and rating games on itch jams.

This bug is noncompliant with WCAG 2.1.'s level A criteria 1.3.1. Info and Relationships: https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships.html
